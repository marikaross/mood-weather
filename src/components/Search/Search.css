.search {
  padding: 30px;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-sizing: border-box;
}

input {
  margin: 5px 0;
  border-radius: 3px;
  border: none;
  height: 30px;
  width: 55%;
  background-color: #f0f0f0;
}

button {
  color: #808080;
  font-size: 1.5rem;
  margin-top: 3px;
  height: 32px;
  border-radius: 3px;
  width: 55%;
  background-color: #f0f0f0;
}

.search input::placeholder {
  font-size: 1.5rem;
  padding-left: 1rem;
  color: #d3d3d3; 
}

