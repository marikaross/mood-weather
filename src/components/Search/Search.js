import React, { Component } from 'react';

export default class Search extends Component {
  constructor(props) {
    super(props)
    this.state = {
      userLat: '',
      userLong: ''
    }
  }

  manageChange = (event) => {
    let {name, value} = event.target
    this.setState({
      [name]: value
    })
  }

  manageSubmit = () => {
    this.props.setLocation(this.state)
    this.setState({
      [name]: ''
    })
  }


  render() {
    return (
      <form className="search" onSubmit={this.manageSubmit}>
        <h1 className="title">Find Your Weather</h1>
        <input 
          type="text" 
          name="userLat"
          placeholder="Enter Latitude" 
          value={this.state.userLat}
          onChange={this.manageChange}
         />
          <input 
          type="text" 
          name="userLong"
          placeholder="Enter Longitude" 
          value={this.state.userLong}
          onChange={this.manageChange}
         />
        <button onClick={(event) => {
          event.preventDefault()
          this.props.fetchWeather(this.state.userInput)
        }}

        >Submit</button> 
      </form>
      )
  }
}