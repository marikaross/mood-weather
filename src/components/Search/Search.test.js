import Search from './Search.js';
import React from 'react';
import { shallow } from 'enzyme';

describe('Search', () => {
  let wrapper,
  mockManageChange,
  mockManageSubmit,
  mockSetLocation,
  mockState;

  beforeEach(() => {
    mockManageChange = jest.fn();
    mockManageSubmit = jest.fn();
    mockSetLocation = jest.fn();
    mockState = {userLat: '', userLong: ''};

    wrapper = shallow(<Search 
      manageChange={mockManageChange}
      manageSubmit={mockManageSubmit}
      setLocation={mockSetLocation}
      state={mockState}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})