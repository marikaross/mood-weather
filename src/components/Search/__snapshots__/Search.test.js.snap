// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    manageChange={[MockFunction]}
    mockManageSubmit={[MockFunction]}
    setLocation={[MockFunction]}
    state={
      Object {
        "userLat": "",
        "userLong": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          name="userLat"
          onChange={[Function]}
          placeholder="Enter Latitude"
          type="text"
          value=""
        />,
        <input
          name="userLong"
          onChange={[Function]}
          placeholder="Enter Longitude"
          type="text"
          value=""
        />,
        <button>
          Submit
        </button>,
      ],
      "className": "search",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "userLat",
          "onChange": [Function],
          "placeholder": "Enter Latitude",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "userLong",
          "onChange": [Function],
          "placeholder": "Enter Longitude",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="userLat"
            onChange={[Function]}
            placeholder="Enter Latitude"
            type="text"
            value=""
          />,
          <input
            name="userLong"
            onChange={[Function]}
            placeholder="Enter Longitude"
            type="text"
            value=""
          />,
          <button>
            Submit
          </button>,
        ],
        "className": "search",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "userLat",
            "onChange": [Function],
            "placeholder": "Enter Latitude",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "userLong",
            "onChange": [Function],
            "placeholder": "Enter Longitude",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
